test/e2e/cloud/nodes.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"time"
+
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("[Feature:CloudProvider][Disruptive] Nodes", func() {
+	f := framework.NewDefaultFramework("cloudprovider")
+	var c clientset.Interface
+
+	BeforeEach(func() {
+		// Only supported in AWS/GCE because those are the only cloud providers
+		// where E2E test are currently running.
+		framework.SkipUnlessProviderIs("aws", "gce", "gke")
+		c = f.ClientSet
+	})
+
+	It("should be deleted on API server if it doesn't exist in the cloud provider", func() {
+		By("deleting a node on the cloud provider")
+
+		nodeDeleteCandidates := framework.GetReadySchedulableNodesOrDie(c)
+		nodeToDelete := nodeDeleteCandidates.Items[0]
+
+		origNodes := framework.GetReadyNodesIncludingTaintedOrDie(c)
+		framework.Logf("Original number of ready nodes: %d", len(origNodes.Items))
+
+		err := framework.DeleteNodeOnCloudProvider(&nodeToDelete)
+		if err != nil {
+			framework.Failf("failed to delete node %q, err: %q", nodeToDelete.Name, err)
+		}
+
+		newNodes, err := framework.CheckNodesReady(c, len(origNodes.Items)-1, 5*time.Minute)
```

Comment:
Sorry for missing this point on my previous review.
Why does here pass the length of origNodes which includes non-schedulable nodes also ?
CheckNodesReady() gets the length of schedulable nodes only and compares it with the parameter size as
https://github.com/kubernetes/kubernetes/blob/ccd83ce1c5daa6d2d1d08ed15f7ae192d594b373/test/e2e/framework/util.go#L4101

So I guessed nodeDeleteCandidates could be valid insteadof origNodes.

Reply:
Ah, I got it. I misunderstood the difference between isNodeSchedulable() and isNodeUntainted() in the framework.

Reply:
Why 5 minutes for `CheckNodeReady`? Is that a magic number that ties to some other timeout, or was that just what seemed reasonable?

Reply:
Yup, that's just what seemed reasonable to me.